core/templates/dev/head/base.js

```diff
@@ -95,6 +97,11 @@ oppia.controller('Base', [
 
     // If this is nonempty, the whole page goes into 'Loading...' mode.
     $rootScope.loadingMessage = '';
+    // Method to call windowSize on resizing the window
+    $window.onresize = function() {
```

Comment:
Hm, am I missing something? Apart from the architectural concerns I mentioned just now, I don't see how using $rootScope is going to get around the overwriting issue (where multiple things in the page want to do something triggered by a resize event, and clobber each others' onresize hooks). Or are you intending to continuously poll the page's width? That can get expensive.

Re calling services, I don't think this is an issue. Having separate services is just a matter of organization, so basically the overhead is in making function calls in the first place, and that's generally negligible. Where you want to be careful is if you're making a bunch of RPCs across the network, but this isn't the case here.